opm/simulators/flow/LogOutputHelper.cpp

```diff
@@ -379,115 +357,80 @@ injection(const std::size_t reportStepNum) const
 
-    const auto& st = summaryState_;
-    for (const auto& gname : schedule_.groupNames()) {
-        auto gName = static_cast<std::string>(gname);
-        auto get = [&st, &gName](const std::string& vector)
-        {
-            const auto key = vector + ':' + gName;
-            return st.has(key) ? st.get(key) : 0.0;
-        };
-
-        tmp_names[0] = gname;
-        if (tmp_names[0] == "FIELD") {
-            tmp_values[2] = st.get("FOIR", 0.0); // WellInjDataType::OilRate
-            tmp_values[3] = st.get("FWIR", 0.0); // WellInjDataType::WaterRate
-            tmp_values[4] = st.get("FGIR", 0.0); // WellInjDataType::GasRate
-            tmp_values[5] = st.get("FVIR", 0.0); // WellInjDataType::FluidResVol
-        } else {
-            tmp_values[2] = get("GOIR"); // WellInjDataType::OilRate
-            tmp_values[3] = get("GWIR"); // WellInjDataType::WaterRate
-            tmp_values[4] = get("GGIR"); // WellInjDataType::GasRate
-            tmp_values[5] = get("GVIR"); // WellInjDataType::FluidResVol
+    const auto& st = this->summaryState_;
+
+    for (const auto& gname : this->schedule_.groupNames()) {
+        std::vector<Scalar> values(WellInjDataType::numWIValues, 0.0);
+        std::vector<std::string> names(WellInjDataType::numWINames, "");
+
+        names[Ix::WellName] = gname;
+
+        if (names[0] == "FIELD") {
+            values[Ix::OilRate]     = st.get("FOIR", 0.0);
+            values[Ix::WaterRate]   = st.get("FWIR", 0.0);
+            values[Ix::GasRate]     = st.get("FGIR", 0.0);
+            values[Ix::FluidResVol] = st.get("FVIR", 0.0);
+        }
+        else {
+            values[Ix::OilRate]     = st.get_group_var(gname, "GOIR", 0.0);
+            values[Ix::WaterRate]   = st.get_group_var(gname, "GWIR", 0.0);
+            values[Ix::GasRate]     = st.get_group_var(gname, "GGIR", 0.0);
+            values[Ix::FluidResVol] = st.get_group_var(gname, "GVIR", 0.0);
         }
 
-        this->outputInjectionReportRecord_(tmp_values, tmp_names);
+        this->outputInjectionReportRecord_(values, names);
     }
 
-    for (const auto& wname : schedule_.wellNames(reportStepNum)) {
-        const auto& well = schedule_.getWell(wname, reportStepNum);
+    for (const auto& wname : this->schedule_.wellNames(reportStepNum)) {
+        const auto& well = this->schedule_.getWell(wname, reportStepNum);
 
-        // Ignore Producer wells
+        // Ignore production wells for the injection report.
         if (well.isProducer()) {
             continue;
         }
 
-        tmp_names[0] = wname; // WellInjDataType::WellName
-        auto wName = static_cast<std::string>(wname);
-        auto get = [&st, &wName](const std::string& vector)
-        {
-            const auto key = vector + ':' + wName;
-            return st.has(key) ? st.get(key) : 0.0;
-        };
+        std::vector<Scalar> values(WellInjDataType::numWIValues, 0.0);
+        std::vector<std::string> names(WellInjDataType::numWINames, "");
```

Comment:
default ctor

Reply:
> default ctor

Agreed.  I've pushed an update to that effect.